iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PrivilegedUserCheck.java

```diff
@@ -0,0 +1,118 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.tree.TreeUtils;
+import org.sonar.iac.docker.tree.api.DockerImageTree;
+import org.sonar.iac.docker.tree.api.DockerTree;
+import org.sonar.iac.docker.tree.api.UserTree;
+
+@Rule(key = "S6471")
+public class PrivilegedUserCheck implements IacCheck {
+
+  private static final Set<String> UNSAFE_IMAGES = Set.of("aerospike", "almalinux", "alpine", "alt", "amazoncorretto",
+    "amazonlinux", "arangodb", "archlinux", "backdrop", "bash", "buildpack-deps", "busybox", "caddy", "cirros", "clearlinux", "clefos",
+    "clojure", "composer", "consul", "couchdb", "crate", "dart", "debian", "drupal", "eclipse-temurin", "elixir", "erlang", "express-gateway",
+    "fedora", "friendica", "gazebo", "gcc", "golang", "haskell", "haxe", "hitch", "httpd", "hylang", "ibmjava", "influxdb", "joomla", "jruby",
+    "julia", "kapacitor", "mageia", "matomo", "maven", "mediawiki", "monica", "mono", "nats", "nats-streaming", "neurodebian", "nextcloud",
+    "nginx", "notary", "openjdk", "oraclelinux", "orientdb", "perl", "photon", "php", "phpmyadmin", "php-zendserver", "plone", "postfixadmin",
+    "pypy", "python", "redmine", "registry", "rethinkdb", "rockylinux", "ros", "ruby", "rust", "r-base", "sapmachine", "satosa", "silverpeas",
+    "sl", "spiped", "swipl", "telegraf", "tomcat", "tomee", "traefik", "ubuntu", "xwiki", "yourls", "bonita", "cassandra", "centos", "chronograf",
+    "convertigo", "couchbase", "docker", "eclipse-mosquitto", "eggdrop", "ghost", "gradle", "mariadb", "mongo", "mongo-express", "mysql", "node",
+    "postgres", "rabbitmq", "rakudo-star", "redis", "sonarqube", "storm", "swift", "teamspeak", "zookeeper");
+  private static final Set<String> UNSAFE_USERS = Set.of("root", "containerAdministrator");
+  private static final String DEFAULT_SAFE_IMAGES = "adminer, api-firewall, elasticsearch, emqx, flink, fluentd, geonetwork, groovy, haproxy,"+
+    " ibm-semeru-runtimes, irssi, jetty, jobber, kibana, kong, lightstreamer, logstash, memcached, neo4j, odoo, open-liberty, percona, "+
+    "rocket.chat, solr, swift, varnish, vault, websphere-liberty, znc, nginxinc/nginx-unprivileged";
+  @RuleProperty(
+    key = "safeImages",
+    description = "Comma separated list of safe images (no default root user).",
+    defaultValue = DEFAULT_SAFE_IMAGES)
+  public String safeImages = DEFAULT_SAFE_IMAGES;
+
+  private Set<String> safeImagesSet;
+  private Set<String> safeImages() {
+    if (safeImagesSet == null) {
+      safeImagesSet = Stream.of(safeImages.split(","))
+        .map(String::trim).collect(Collectors.toSet());
```

Comment:
when a user provides safe images, shouldn't we merge this provided list with the `DEFAULT_SAME_IMAGES`? 
I guess for the default ones, we're sure they aren't problematic regardless of the provided parameter.

Reply:
Sounds good to me, I wasn't sure if the provided list should be a default editable list or not, but indeed we are normally sure they are safe so the user should only be able to add their safe images.
Will update this.
I will also modify the code in case the user put an unsafe image in their personalized safe list to not report an error.